service/history/outbound_queue_standby_task_executor.go

```diff
@@ -75,14 +93,152 @@ func (e *outboundQueueStandbyTaskExecutor) Execute(
 		metrics.OperationTag(taskType),
 	}
 
-	if err == nil {
-		smRegistry := e.shardContext.StateMachineRegistry()
-		err = smRegistry.ExecuteStandbyImmediateTask(ctx, e, ref, smt)
+	if err != nil {
+		return queues.ExecuteResponse{
+			ExecutionMetricTags: metricsTags,
+			ExecutedAsActive:    false,
+			ExecutionErr:        err,
+		}
+	}
+
+	actionFn := func(ctx context.Context) (any, error) {
+		err := e.Access(ctx, ref, hsm.AccessRead, func(node *hsm.Node) error {
+			if smt.Concurrent() {
+				//nolint:revive // concurrent tasks implements hsm.ConcurrentTask interface
+				concurrentSmt := smt.(hsm.ConcurrentTask)
```

Comment:
This looks okay to me but in `TaskGenerator` we should panic if a concurrent task was scheduled that doesn't implement `Validate`. Make sure the panic message is helpful to developers using the framework.

Reply:
How about we change the `RegisterExecutor` to get a instance of the task instead of just the task type id, and make the verification/panic there instead of during `TaskGenerator`? I feel like panic during initialization of server better than panicking during task generation.

Reply:
That would be nice but I feel like it would complicate registration. Feel free to try that out and see how it works out.